gravitational/teleport
# set cluster connector name on signin for first cloud user

### What

Automatically set the passwordless cluster authentication preference for cloud users.

When the first user chooses to sign in with passwordless, `cluster_auth_preference.connector_name` will be set to `"passwordless"`. This will instruct login and sign-in flow to use `passwordless` as the default method (the user can still choose another available method).


### Testing guide:
1. checkout to the branch (make sure teleport.e is cloned)
2. Create a new tenant in the staging sales-center
3. build a dev version of teleport-ent and use it in the staging tenant (see https://github.com/gravitational/teleport.e/blob/master/dev-deploy.md for exact steps) 
_If you run into a missing `DEB_PATH` error you can temporarily restore `build.assets/charts/Dockerfile` old code before https://github.com/gravitational/teleport/pull/17597/files#diff-3b89b68ec0c7abb15cbf57af77d2f3cced352cbe65159772d3b680ef76f9915e_
4. Go to the invite link created in step 2. and use the passwordless form to authenticate
5. In Web UI create a second user.
6. After entering an invite link the second user should see the passwordless form initially instead of the password + TOTP form

Implements RFD: https://github.com/gravitational/cloud/pull/2405
Fixes: https://github.com/gravitational/cloud/issues/2069
